vscode-extension/src/extension.ts

```diff
@@ -1,99 +1,144 @@
 /*---------------------------------------------------------
  * Copyright (C) Microsoft Corporation. All rights reserved.
  *--------------------------------------------------------*/
-// import * as vscode from 'vscode';
-"use strict";
-
-import path = require("path");
-import vscode = require("vscode");
-import TelemetryReporter from "vscode-extension-telemetry";
-import { DocumentSelector } from "vscode-languageclient";
-import { CodeActionsFeature } from "./features/CodeActions";
-import { Logger, LogLevel } from "./logging";
-import { SessionManager } from "./session";
-import Settings = require("./settings");
-import { PowerShellLanguageId } from "./utils";
-import { LanguageClientConsumer } from "./languageClientConsumer";
-import { getSrcVersion } from "./selectVersion";
-
-// The most reliable way to get the name and version of the current extension.
-// tslint:disable-next-line: no-var-requires
-const PackageJSON: any = require("../package.json");
-
-// the application insights key (also known as instrumentation key) used for telemetry.
-
-let logger: Logger;
-let sessionManager: SessionManager;
-let languageClientConsumers: LanguageClientConsumer[] = [];
-let commandRegistrations: vscode.Disposable[] = [];
-let telemetryReporter: TelemetryReporter;
 
-const documentSelector: DocumentSelector = [
-    { language: "powershell", scheme: "file" },
-    { language: "Powershell", scheme: "file" },
-    { language: "powershell", scheme: "untitled" },
-];
+// The module 'vscode' contains the VS Code extensibility API
+// Import the module and reference it with the alias vscode in your code below
+import * as vscode from 'vscode';
+import { BreakingChangeInfo } from './quickFix';
+import { updateDiagnostics } from './diagnostic';
+import {
+    getPlatformDetails, IPlatformDetails, IPowerShellExeDetails,
+    OperatingSystem, PowerShellExeFinder
+} from "./platform";
+import { Logger } from "./logging";
+import { PowershellProcess } from './powershell';
 
-export function activate(context: vscode.ExtensionContext): void {
-    console.debug('"azps-tools" is activating ...');
-
-	// The command has been defined in the package.json file
-	// Now provide the implementation of the command with registerCommand
-	// The commandId parameter must match the command field in package.json
+const PackageJSON: any = require("../package.json");
+let powershell = new PowershellProcess();
 
-    // Create the logger
-    logger = new Logger();
+// this method is called when your extension is activated
+// your extension is activated the very first time the command is executed
+export async function activate(context: vscode.ExtensionContext) {
```

Comment:
Instead of a long function, you should group some logic and turn them into smaller functions